app/controllers/account_requests_controller.rb

```diff
@@ -0,0 +1,56 @@
+class AccountRequestsController < ApplicationController
+  skip_before_action :authorize_user
+  skip_before_action :authenticate_user!
+
+  before_action :set_account_request_from_token, only: [:received, :confirmation, :confirm]
+
+  layout 'devise'
+
+  def received; end
+
+  def confirmation; end
+
+  def confirm
+    @account_request.update!(confirmed_at: Time.current)
+    AccountRequestMailer.approval_request(account_request_id: @account_request.id).deliver_later
+  end
+
+  def invalid_token; end
+
+  def new
+    @account_request = AccountRequest.new
+  end
+
+  def create
+    @account_request = AccountRequest.new(account_request_params)
+
+    if @account_request.save
+      AccountRequestMailer.confirmation(account_request_id: @account_request.id).deliver_later
+
+      redirect_to received_account_requests_path(token: @account_request.identity_token),
+                  notice: 'Account request was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_account_request_from_token
+    @account_request = AccountRequest.get_by_identity_token(params[:token])
+
+    # Use confirmation timestamp instead
+    if @account_request.nil? || @account_request.confirmed? || @account_request.processed?
+      redirect_to invalid_token_account_requests_path(token: params[:token])
+      return
+    end
```

Comment:
If you had the `get_by_identity_token` integrate `confirmed_at` and `organization` into its `find_by!`, it would raise a `RecordNotFound`, and you could rescue those `RecordNotFound` in this controller and handle it by doing the `redirect_to ...`. 

Maybe in practice that would be worse -- just an idea that popped up. I really like that you made `AccountRequest` be an actual resource, and the JWT seems to function like the lookup param. I *think* that leveraging these behaviors into the typical behaviors for a Rails model might make the intent clearer?

Reply:
> If you had the get_by_identity_token integrate confirmed_at and organization into its find_by!, it would raise a RecordNotFound, and you could rescue those RecordNotFound in this controller and handle it by doing the redirect_to ....

Hmm, I think that could work but then I think it is unclear why a method like `get_by_identity_token` didn't return something. In this case, I think the controller should be responsible for redirecting.

> Maybe in practice that would be worse -- just an idea that popped up. I really like that you made AccountRequest be an actual resource, and the JWT seems to function like the lookup param. I think that leveraging these behaviors into the typical behaviors for a Rails model might make the intent clearer?

Hmm yeh, I think treating it more like a resource was my intent at the start. However, the JWT tokens for me felt like it should operate differently to avoid getting it confused for primary keys and traditional ways of handling resources :)